iso15118/secc/states/din_spec_states.py

```diff
@@ -644,14 +644,19 @@ async def process_message(
                 "Stay in this state and expect "
                 "WeldingDetectionReq/SessionStopReq"
             )
+            await self.comm_session.evse_controller.set_hlc_charging(False)
             next_state = None
-            await self.comm_session.evse_controller.open_contactor()
+
```

Comment:
I understand you were looking at the closing contactor side of things, but I see contactor is not closed in din_spec. Do you want to paste the below lines under the `if power_delivery_req.ready_to_charge:` block:
```
            contactor_state = await self.comm_session.evse_controller.close_contactor()
            if contactor_state != Contactor.CLOSED:
                self.stop_state_machine(
                    "Contactor didn't close",
                    message,
                    ResponseCode.FAILED,
                )
                return
```

Reply:
It actually has to happen during CableCheck in the DC scenario... I will add it there, for both -2 and Din